src/server/cmd/pachd/main.go

```diff
@@ -61,15 +64,110 @@ type appEnv struct {
 	BlockCacheBytes       string `env:"BLOCK_CACHE_BYTES,default=5G"`
 	PFSCacheBytes         string `env:"PFS_CACHE_BYTES,default=1G"`
 	WorkerImage           string `env:"WORKER_IMAGE,default="`
+	WorkerSidecarImage    string `env:"WORKER_SIDECAR_IMAGE,default="`
 	WorkerImagePullPolicy string `env:"WORKER_IMAGE_PULL_POLICY,default="`
 	LogLevel              string `env:"LOG_LEVEL,default=info"`
 }
 
 func main() {
-	cmdutil.Main(do, &appEnv{})
+	switch mode {
+	case "full":
+		cmdutil.Main(doFullMode, &appEnv{})
+	case "pfs":
+		cmdutil.Main(doPFSMode, &appEnv{})
+	default:
+		fmt.Println("unrecognized mode: %s", mode)
+	}
+}
```

Comment:
It'd be cool if `doFullMode` and `doPFSMode` were a bit more composable / duplicated less code. It seems like they both have in common that they both instantiate some servers and ultimately register them as part of `grpcutil.Serve`. What if you made a function like `PFSServers(appEnvObj interface{}) func(s *grpc.Server)` that returns a function which registers the servers it instantiated. It seems like that way you could pretty easily compose whatever behavior you want.